VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/non-host-waiting-room/non-host-waiting-room.component.ts

```diff
@@ -0,0 +1,435 @@
+import { take, takeUntil } from 'rxjs/operators';
+import { UnloadDetectorService } from 'src/app/services/unload-detector.service';
+import { UserMediaService } from 'src/app/services/user-media.service';
+import { ParticipantMediaStatus } from 'src/app/shared/models/participant-media-status';
+import { Title } from '@angular/platform-browser';
+import { ModalTrapFocus } from '../../shared/modal/modal-trap-focus';
+import { HideComponentsService } from '../services/hide-components.service';
+import { FocusService } from 'src/app/services/focus.service';
+import { ConferenceState } from '../store/reducers/conference.reducer';
+import { Store } from '@ngrx/store';
+import { LaunchDarklyService } from 'src/app/services/launch-darkly.service';
+import { VHParticipant } from '../store/models/vh-conference';
+import { NonHostUserRole } from '../waiting-room-shared/models/non-host-user-role';
+
+@Component({
+    standalone: false,
+    selector: 'app-non-host-waiting-room',
+    templateUrl: './non-host-waiting-room.component.html',
+    styleUrls: ['../waiting-room-global-styles.scss', './non-host-waiting-room.component.scss']
+})
+export class NonHostWaitingRoomComponent extends WaitingRoomBaseDirective implements OnInit, OnDestroy {
+    @Input() userRole: NonHostUserRole;
+
+    currentTime: Date;
+    hearingStartingAnnounced: boolean;
+
+    isParticipantsPanelHidden = false;
+    hearingVenueIsScottish$: Observable<boolean>;
+
+    showWarning = false;
+    displayLanguageModal: boolean;
+    displayLeaveHearingPopup = false;
+
+    private readonly componentLoggerPrefix = '[Non-Host Waiting Room] -';
+    private destroyedSubject = new Subject();
+    private titleForParticipant = 'Participant waiting room';
+    private titleForJoh = 'JOH waiting room';
+
+    constructor(
+        protected route: ActivatedRoute,
+        protected videoWebService: VideoWebService,
+        protected eventService: EventsService,
+        protected logger: Logger,
+        protected errorService: ErrorService,
+        protected videoCallService: VideoCallService,
+        protected deviceTypeService: DeviceTypeService,
+        protected router: Router,
+        protected consultationService: ConsultationService,
+        protected notificationSoundsService: NotificationSoundsService,
+        protected notificationToastrService: NotificationToastrService,
+        protected roomClosingToastrService: RoomClosingToastrService,
+        protected clockService: ClockService,
+        protected translateService: TranslateService,
+        protected consultationInvitiationService: ConsultationInvitationService,
+        private unloadDetectorService: UnloadDetectorService,
+        protected userMediaService: UserMediaService,
+        protected titleService: Title,
+        protected hideComponentsService: HideComponentsService,
+        protected focusService: FocusService,
+        protected launchDarklyService: LaunchDarklyService,
```

Comment:
can the `FeatureFlagDirective` achieve the same result instead of adding a dependency to manage